Fix #3976 - Add 'Minimum Outdoor Dry-Bulb Temperature for Compressor Operation' to all Coil:Cooling:DX:XXX objects

jmarrec opened the pull request:
Pull request overview
---------------------

 - Fix #3976
 - Add 'Minimum Outdoor Dry-Bulb Temperature for Compressor Operation' to all Coil:Cooling:DX:XXX objects:
    * `OS:Coil:Cooling:DX:SingleSpeed`
    * `OS:Coil:Cooling:DX:MultiSpeed`
    * `OS:Coil:Cooling:DX:VariableSpeed`
    * `OS:Coil:Cooling:DX:TwoStageWithHumidityControlMode`
    * `OS:Coil:Cooling:DX:TwoSpeed`
        * This one also gained the field `Unit Internal Static Air Pressure`

Added:

* IDD Changes
* Model setters getters resetters and isXXDefaulted
* Model Tests
* FT handling
* FT Tests
* RT Handling only for Coil:Cooling:DX:SingleSpeed where it existed (and no tests possible since even for this object it's actually disabled in the RT)
* VersionTranslation
* VersionTranslation Tests

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [x] Model API Changes / Additions
 - [x] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [x] Model API methods are tested (in `src/model/test`)
 - [x] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [x] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [x] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

jmarrec commented on resources/model/OpenStudio.idd:
```diff
@@ -15103,7 +15103,11 @@ OS:Coil:Cooling:DX:SingleSpeed,
        \required-field
        \object-list QuadraticCubicCurves
-  N6, \field Nominal Time for Condensate Removal to Begin
+  N6 , \field Minimum Outdoor Dry-Bulb Temperature for Compressor Operation
+       \type real
+       \default -25.0
+       \units C
```
`OS:Coil:Cooling:DX:SingleSpeed`. 

**The exact same (real, default -25C, units C) is used for all 5 coils.**

**This could be changed to `\required-field`, the default moved to the Ctor (`this->setMinimumOutdoorDryBulbTemperatureforCompressorOperation(-25)`), and remove the `isMinimumOutdoorDryBulbTemperatureforCompressorOperationDefaulted()` and `resetMinimumOutdoorDryBulbTemperatureforCompressorOperation()`**

@kbenne please advise.

kbenne replied:
Yes please. I prefer required fields and initialize in the ctor.

jmarrec commented on resources/model/OpenStudio.idd:
```diff
@@ -15281,7 +15285,11 @@ OS:Coil:Cooling:DX:TwoStageWithHumidityControlMode,
        \type object-list
        \object-list WaterStorageTankNames
-  N5,  \field Basin Heater Capacity
+  N5,  \field Minimum Outdoor Dry-Bulb Temperature for Compressor Operation
+       \type real
+       \default -25.0
+       \units C
```
OS:Coil:Cooling:DX:TwoStageWithHumidityControlMode

jmarrec commented on resources/model/OpenStudio.idd:
```diff
@@ -15523,6 +15531,10 @@ OS:Coil:Cooling:DX:MultiSpeed,
        \key AirCooled
        \key EvaporativelyCooled
        \required-field
+  N1 , \field Minimum Outdoor Dry-Bulb Temperature for Compressor Operation
+       \type real
+       \default -25.0
+       \units C
```
OS:Coil:Cooling:DX:MultiSpeed

jmarrec commented on resources/model/OpenStudio.idd:
```diff
@@ -15734,6 +15746,15 @@ OS:Coil:Cooling:DX:TwoSpeed,
        \autosizable
        \units m3/s
        \minimum> 0
+  N5 , \field Unit Internal Static Air Pressure
+       \note Enter pressure drop for the unit containing the coil.
+       \note This value is only used to calculate Energy Efficiency Ratio
+       \note (EER), Integrated Energy Efficiency Ratio (IEER), and the
+       \note Standard Rating (Net) Cooling Capacity. This value is not
+       \note used for modeling the evaporator fan during simulations.
+       \type real
+       \units Pa
+       \minimum> 0.0
```
OS:Coil:Cooling:DX:TwoSpeed

This could be changed to be required-field, and ctor assign the apparently-used default listed in the I/O ref https://bigladdersoftware.com/epx/docs/9-3/input-output-reference/group-heating-and-cooling-coils.html#field-unit-internal-static-air-pressure

> If this field is not used, then the standard ratings are still performed but use a default for specific fan power of **773.3 (W/(m3/s))**

@kbenne  please advise

kbenne replied:
I also prefer required field and initialize in ctor in this case.

jmarrec commented on resources/model/OpenStudio.idd:
```diff
@@ -15833,37 +15854,41 @@ OS:Coil:Cooling:DX:TwoSpeed,
        \key AirCooled
        \key EvaporativelyCooled
-  N9, \field High Speed Evaporative Condenser Effectiveness
+  N10, \field Minimum Outdoor Dry-Bulb Temperature for Compressor Operation
+       \type real
+       \default -25.0
+       \units C
```
OS:Coil:Cooling:DX:TwoSpeed

jmarrec commented on resources/model/OpenStudio.idd:
```diff
@@ -17571,18 +17596,22 @@ OS:Coil:Cooling:DX:VariableSpeed,
         \type real
         \minimum 0.0
         \units C
+   N9,  \field Minimum Outdoor Dry-Bulb Temperature for Compressor Operation
+        \type real
+        \default -25.0
+        \units C
```
OS:Coil:Cooling:DX:VariableSpeed

jmarrec commented:
Rebased onto develop and force-pushed, since this PR included the bump version to 301 PR that was merged in. The diff is much cleaner now (30000 lines were just committing Openstudio.idd for 3.0.0 official)

I addressed your review comments @kbenne and made everything a required-field.
